### PR TITLE
Add Windows support to lit.cfg

### PR DESCRIPTION
**Pull request by kaadam**
* Enable running Flang tests on Windows

### CODE REVIEW

**Review by bryanpkc — APPROVED** [review]
LGTM

**Review by shivaramaarao — APPROVED** [review]

File: test/lit.cfg [hunk 1]
```diff
@@ -27,10 +27,6 @@ import lit.llvm
 lit.llvm.initialize(lit_config, config)
 lit.llvm.llvm_config.with_system_environment("COMPILER_PATH")
 
-if platform.system() == 'Windows':
```
Comment (pawosm-arm) [inline test/lit.cfg:30]: The comments below refers to Msys (MinGW companion), what about Cygwin or VisualStudio, are we compatible with them so we can drop this condition?

Comment (kaadam) [reply test/lit.cfg:30]: I've run tests from Windows environment (PowerShell, Dev Command Prompt) so basically the configuration itself is compatible with Windows /MS Visual Studio/. I didn't try it under Cygwin or MSYS envs. It should work. However we use several gnu tools to support Flang test running on Windows such 'bash, sh'  (from Git for Windows /mingw64/) or make and others from 'gnuwin32' package. 
Shall I add a reminder comment for that, or change the condition to exclude 'cygwin'?

Comment (pawosm-arm) [reply test/lit.cfg:30]: A comment seems sufficient

Comment (kaadam) [reply test/lit.cfg:30]: @pawosm-arm What do you think about the comment?

Comment (pawosm-arm) [reply test/lit.cfg:30]: I'd rather put MS Visual Studio (full name), so it won't make confusion with MSVC or MSYS.

Comment (kaadam) [reply test/lit.cfg:30]: okay, updated.

**Review by pawosm-arm — APPROVED** [review]

**Pull request merged**